File: JSPM.Web/Modules/ServerTypes/Organisation/InstituteForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface InstituteForm {
    InstituteName: StringEditor;
    Description: StringEditor;
}

export class InstituteForm extends PrefixedContext {
    static formKey = 'Organisation.Institute';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!InstituteForm.init)  {
            InstituteForm.init = true;

            var w0 = StringEditor;

            initFormType(InstituteForm, [
                'InstituteName', w0,
                'Description', w0
            ]);
        }
    }
}

File: JSPM.Web/Modules/Organisation/Institute/InstituteForm.cs
using Serenity.ComponentModel;

namespace JSPM.Organisation.Forms;

[FormScript("Organisation.Institute")]
[BasedOnRow(typeof(InstituteRow), CheckNames = true)]
public class InstituteForm
{
    public string InstituteName { get; set; }
    public string Description { get; set; }
}